### PR TITLE
ZA_2-Marvin_Marshall-JS-Core-1-WK1

### PR DESCRIPTION
**Pull request by MarvinLeeMarshall**
I have passed all test in the JS files.

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered exercises/B-hello-world/README.md](https://github.com/MarvinLeeMarshall/JavaScript-Core-1-Coursework-Week1/blob/master/exercises/B-hello-world/README.md)

### CODE REVIEW

**Comment by Getson47** [comment]
Keep pushing!

File: exercises/I-floats/exercise.js [hunk 1]
```diff
@@ -1,2 +1,18 @@
 var numberOfStudents = 15;
 var numberOfMentors = 8;
+
+var percentSign = "%"
+var total = numberOfStudents + numberOfMentors;
+/*calculation for students*/
+
+var text1 = "Percentage of Students ";
```
Comment (chezlinbenson) [inline exercises/I-floats/exercise.js:8]: Great understanding of basic javascript syntax and naming conventions. No need to understand all the words, as long as you understand the concepts.

File: exercises/K-functions-parameters/exercise5.js [hunk 2]
```diff
@@ -1,4 +1,7 @@
 // Declare your function here
+function createLongGreeting(name, age){
```
Comment (chezlinbenson) [inline exercises/K-functions-parameters/exercise5.js:2]: Understanding string concaternation takes basic exercises like these. You did well in this exercise analysing your code and code stucture.

File: exercises/L-functions-nested/exercise.js [hunk 3]
```diff
@@ -3,3 +3,23 @@ var mentor2 = "Irina";
 var mentor3 = "Mimi";
 var mentor4 = "Rob";
 var mentor5 = "Yohannes";
+
+
+function shoutyGreeting(name){
```
Comment (chezlinbenson) [inline exercises/L-functions-nested/exercise.js:8]: This exercise had many trainees get to use more of their brains. There are many solutions to any code. Your basic functions show great understanding on Declaring Functions and returning them. Well done!

File: mandatory/3-function-output.js [hunk 4]
```diff
@@ -3,12 +3,15 @@ function getRandomNumber() {
   return Math.random() * 10;
 }
 
+//The above function grabs a random number by using the Math.random() method and multiplies that random number by 10.
```
Comment (chezlinbenson) [inline mandatory/3-function-output.js:6]: Very good comments. They are clear and uses the Javascript proper commenting.

File: mandatory/4-tax.js [hunk 5]
```diff
@@ -17,7 +21,13 @@ function calculateSalesTax() {}
   Remember that the prices must include the sales tax (hint: you already wrote a function for this!)
 */
 
-function addTaxAndFormatCurrency() {}
+function addTaxAndFormatCurrency(price){
```
Comment (chezlinbenson) [inline mandatory/4-tax.js:24]: Very clean code, well done! There are various options to the functions you are coding to make it shorter. Making use of the correct methods in your code, are best practise, Well done there!

**Review by chezlinbenson — COMMENTED** [review]
These exercises were good challenges to understand the basic Javascript language. declaring functions, variables, parameters, arguments and naming conventions, were the goals for week1, and you have shown great skills in the javascript basic language. Well Done! Continue following the basic structures and concepts moving forward.

File: mandatory/1-syntax-errors.js [hunk 6]
```diff
@@ -1,18 +1,17 @@
...
-  return "Hello, my name is " + name "and I am " age + "years old";
-
+function introduceMe(name, age){
+  return "Hello, my name is " + name + " and I am " + age + " years old";
+}
 function getTotal(a, b) {
-  total = a ++ b;
-
-  return "The total is total";
+  return "The total is " + (a + b);
```
Comment (riaanduplessis) [inline mandatory/1-syntax-errors.js:11]: Good call using the brackets here `+ (a + b)` 👌

**Review by riaanduplessis — COMMENTED** [review]
Awesome work Marvin! 🎉

**Pull request closed without merging**